### PR TITLE
CIRCSTORE-429: Validate allowed service points on POST and PUT

### PR DESCRIPTION
**Pull request by OleksandrVidinieiev**
Do not allow saving or updating a request policy when at least one of the configured allowed pickup service points does not exist or is not explicitly marked as a pickup location.
Resolves [CIRCSTORE-429](https://issues.folio.org/browse/CIRCSTORE-429).

### CODE REVIEW

File: src/main/java/org/folio/rest/impl/RequestPoliciesAPI.java [hunk 1]
```diff
@@ -59,43 +66,58 @@ public void postRequestPolicyStorageRequestPolicies(String lang, RequestPolicy e
...
-                    io.vertx.core.Future.succeededFuture(
-                      RequestPolicyStorage.PostRequestPolicyStorageRequestPoliciesResponse
-                        .respond500WithTextPlain(reply.cause().toString())));
-                }
-              } catch (Exception e) {
-                log.error(e);
-                asyncResultHandler.handle(
-                  io.vertx.core.Future.succeededFuture(
-                    RequestPolicyStorage.PostRequestPolicyStorageRequestPoliciesResponse
-                      .respond500WithTextPlain(e.getMessage())));
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/rest/impl/RequestPoliciesAPI.java:86]: This block of code was moved under `.onSuccess()` with no changes

File: ramls/service-point.json [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  "description": "A service point",
+  "type": "object",
+  "properties": {
+    "id": {
+      "type": "string",
+      "description": "Id of service-point object"
+    },
+    "name": {
+      "type": "string",
+      "description" : "service-point name, a required field"
```
Comment (roman-barannyk) [inline ramls/service-point.json:12]:
```suggestion
      "description": "service-point name, a required field"
```

File: ramls/service-point.json [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    },
+    "code": {
+      "type": "string",
+      "description" : "service-point code, a required field"
```
Comment (roman-barannyk) [inline ramls/service-point.json:16]:
```suggestion
      "description": "service-point code, a required field"
```

File: ramls/service-point.json [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    },
+    "discoveryDisplayName": {
+      "type": "string",
+      "description": "display name, a required field"
+    },
+    "description": {
+      "type": "string",
+      "description" : "description of the service-point"
```
Comment (roman-barannyk) [inline ramls/service-point.json:24]:
```suggestion
      "description": "description of the service-point"
```

File: ramls/time-period.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "description" : "schema for time-period, which contains time interval 'duration' and the time unit",
```
Comment (roman-barannyk) [inline ramls/time-period.json:4]:
```suggestion
  "description": "schema for time-period, which contains time interval 'duration' and the time unit",
```

File: ramls/time-period.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      "type": "string",
+      "description": "Unit of time for the duration",
+      "enum":[
+        "Minutes",
+        "Hours",
+        "Days",
+        "Weeks",
+        "Months"
+      ],
+      "default" : "Days"
```
Comment (roman-barannyk) [inline ramls/time-period.json:20]:
```suggestion
      "default": "Days"
```

File: src/main/java/org/folio/service/policy/RequestPolicyValidationService.java [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    for (String id : allowedServicePointIds) {
+      Servicepoint servicePoint = servicePointsById.get(id);
+      if (servicePoint == null) {
+        errors.add(new Error()
+          .withMessage(String.format("Service point %s does not exist", id))
+          .withParameters(singletonList(new Parameter().withKey("servicePointId").withValue(id))));
+      } else if (servicePoint.getPickupLocation() == null || !servicePoint.getPickupLocation()) {
+        errors.add(new Error()
+          .withMessage(String.format("Service point %s is not a pickup location", id))
+          .withParameters(singletonList(new Parameter().withKey("servicePointId").withValue(id))));
```
[the remark this answers is not in the document]
Comment (OleksandrVidinieiev) [reply src/main/java/org/folio/service/policy/RequestPolicyValidationService.java]: Done

**Review by alexanderkurash — APPROVED** [review]

**Review by roman-barannyk — APPROVED** [review]

**Pull request merged**